src/main/scala/com/microsoft/hyperspace/index/FileBasedSignatureProvider.scala

```diff
@@ -34,19 +34,25 @@ class FileBasedSignatureProvider extends LogicalPlanSignatureProvider {
    * Generate the signature of logical plan.
    *
    * @param logicalPlan logical plan of data frame.
+   * @param targetFiles list of file paths used to calculate signature
    * @return signature, if the logical plan has some LogicalRelation operator(s); Otherwise None.
    */
-  def signature(logicalPlan: LogicalPlan): Option[String] = {
-    fingerprintVisitor(logicalPlan).map(HashingUtils.md5Hex)
+  override def signature(
+      logicalPlan: LogicalPlan,
+      targetFiles: Option[Set[Path]]): Option[String] = {
```

Comment:
Question: Is there a specific reason that `Set` is used here? Like enforcing some uniqueness constraint? (because we normally use Seq)